### ISSUE TITLE
mne.grand_average for source estimates

### ISSUE THREAD

**Issue by CarinaFo** [mne-tools/mne-python#12109]
### Describe the new feature or enhancement

For consistency between sensor and source analysis, it would be nice to allow for [mne.grand_average](https://mne.tools/stable/generated/mne.grand_average.html) as input data a list of [mne.SourceEstimate](https://mne.tools/stable/generated/mne.SourceEstimate.html). Users might be fooled into believing that grand avering source estimates it not suggested.

### Describe your proposed implementation

the all_inst parameter that takes the evoked datasets of [mne.grand_average](https://mne.tools/stable/generated/mne.grand_average.html) should  allow for list of [mne.SourceEstimate](https://mne.tools/stable/generated/mne.SourceEstimate.html). All source estimates in the list (https://mne.tools/stable/generated/mne.SourceEstimate.html) should have the same number of vertices. The nave parameter of the evokeds could be set to 1 as a workaround. interpolate_bads and drop_bads should check if evokeds equal Source Estimate and if True, both parameters should be set to False.

### Describe possible alternatives

An alternative could be to implement a separate function for the source estimate class  [mne.SourceEstimate], that computes grand_averages on a list of source estimates.

**Comment by drammock** [comment]
just out of curiosity, what is the motivation for this versus doing the grand average on Evoked sensor data before applying the inverse?  Are you doing some additional processing of the STCs before averaging?

**Comment by CarinaFo** [comment]
Good point, there is a note in the documentation of [mne.grand_average](https://mne.tools/stable/generated/mne.grand_average.html) that states: "A grand average evoked should not be used for source localization." There is no further explanation why this is a bad idea.

**Comment by hoechenberger** [comment]
It probably has to do with the way we whiten the data? The thought of an average noise covariants matrix raises my internal eyebrows a bit ... but I'm just blindly guessing here! @agramfort probably knows the answer!

**Comment by drammock** [comment]
Right, of course, I wasn't thinking of the across-subjects case (but of course that's the main use case for grand averages). Another problem is that at the sensor level we have no guarantee of equivalent head position (and for EEG, no guarantee of equivalent *sensor* position even). After inverse and morphing those problems go away
